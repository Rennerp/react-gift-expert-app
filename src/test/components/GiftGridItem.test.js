import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

describe('Debe reenderizar el componente GiftGridItem', () => {

  const title = 'Un Titulo';
  const url = 'https://localhost/algo.jpg'
  let wrapper = shallow(<GifGridItem title = { title } url = {url} />);

  test('Reenderiza el componente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Verificacion del titulo', () => {
    const titulo = wrapper.find('p').text().trim();
    expect(titulo).toBe(title);
  })

  test('Verificando la imagen', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  })

  test('Verificando la clase del div', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  })

})
