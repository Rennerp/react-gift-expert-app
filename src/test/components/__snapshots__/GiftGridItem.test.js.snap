// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe reenderizar el componente GiftGridItem Reenderiza el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
